### DISCUSSION TITLE
Sound on slider sometimes don't played due to short amount of beat spacing in edit mode.

### DISCUSSION THREAD

**Discussion by timelessnesses** [ppy/osu#16335] [Q&A] [answered]
(This only works with slider with slider only!)  
Slider head sound will not be played sometimes if the beat gap is small.  
Beatmap Example:  
https://osu.ppy.sh/beatmaps/3402199
The top difficulty (Whistle's Extra) the slider head sound sometimes doesn't played if the beat gap is way too small.  
This can also be occured with bad laptop/pc too.

**Comment by peppy** [comment]
Can you confirm that you're only seeing this in the editor?

**Reply by bdach** [reply]
also please provide a timestamp because i can't hear what you mean on checking.

**Reply by timelessnesses** [reply]
> Can you confirm that you're only seeing this in the editor?

yes

**Reply by timelessnesses** [reply]
> also please provide a timestamp because i can't hear what you mean on checking.

every slider that have 1/8 gap like  
00:09:264 (1,2,3,4,1,1,2,3,4,5,6,1,1,2,1,2,3,1,2,1,2,1,2,3,1,2,1,1,1,1,2,3,4,1,2,3,1,2,3,1,2,1,2,3,4,5)  
00:32:121 (1,2,3,4,1,1,2,3,4,5,6,1,1,2,1,2,1,1,2,3,4,5,6,1,1,2,1,2,1,1,2,3,4,1,1,1,1,2,3,4,1,1,2,3,4,1,1)  
and sometimes if slider wasn't long enough and you have bad pc you will not hear slider end sound in editor like  
00:21:941 (1,2,1,2,3,4,5,6,7,8,1,2,3,1,2,1,2,3,4,1,2,1,1,1,1,1,1,1,1) -

**Reply by bdach** [reply]
these sliders in particular look to have intentionally-set 5% volume sample control points at their ends.

![2022-01-07-164422_604x443_scrot](https://user-images.githubusercontent.com/20418176/148568539-5cbe1bef-e0ac-40db-b8dc-e3ef1a30dbbb.png)

so i'm not sure still what the issue is here...?

**Reply by timelessnesses** [reply]
> these sliders in particular look to have intentionally-set 5% volume sample control points at their ends.
> 
> ![2022-01-07-164422_604x443_scrot](https://user-images.githubusercontent.com/20418176/148568539-5cbe1bef-e0ac-40db-b8dc-e3ef1a30dbbb.png)
> 
> so i'm not sure still what the issue is here...?

The issue is slider head sound sometimes doesn't played in edit on osu lazer

**Reply by peppy** [reply]
What frame rate is your editor running at to reproduced this issue? If it's very low frames and very short sliders, there's change it will never play them due to non-frame-stable playback (an optimisation to make the editor faster).

**Reply by timelessnesses** [reply]
> What frame rate is your editor running at to reproduced this issue? If it's very low frames and very short sliders, there's change it will never play them due to non-frame-stable playback (an optimisation to make the editor faster).

15 - 30

**Reply by peppy** [reply]
that's going to be the issue for sure. if you're not running multi threaded could you try to change that and see if it helps at all?

**Reply by timelessnesses** [reply]
yeah it helped a lot

**Comment by timelessnesses** [comment, marked as answer]
Resolved with multi threading and vsync fps cap
